docs/components/aws/elasticache.md

```diff
@@ -0,0 +1,792 @@
+
+##### Using Existing VPC
+
+To use already existing VPC, select **Existing VPC** option from VPC type under VPC settings.
+
+<p align="center">
+  <img style="Padding:20px" width="400" src="./images/elasticache/ecache_existing_vpc.png">
+</p>
+
+Field          | Required           | Description
+---            | :---:              | ---
+VPC            | :white_check_mark: | Select a VPC from available VPC dropdown 
+Subnet         | :white_check_mark: | Select a subnet from available subnets for selected VPC 
+Security Group | :white_check_mark: | Select a security group from available security groups for selected VPC 
+
+**Important:**  
+> **When using an existing VPC, make sure your lambda function is also deployed in the same VPC. Either select the same VPC 
+(with the same Subnet and Security Group) that you selected for ElastiCache cluster in lambda advanced configuration or 
+leave 'No Vpc' option so that Sigma IDE will be able to deployed your lambda function in the same VPC as ElastiCache cluster**
+
+### Existing Redis Cluster
+
+To choose an existing redis cluster, go to the **Existing Cluster** tab of the configuration panel. Existing cluster 
+instances can be selected from Existing cluster dropdown.
+
+<p align="center">
+  <img style="Padding:20px" width="400" src="./images/elasticache/ecache_existing_cluster.png">
+</p>
+
+### Basic Code Template
+
+After ElastiCache configuration completed, Sigma IDE will generate following basic code template for you. 
+
+```
+// You must always quit the redis client after it's used
+redis.[operation-name]({
+    clusterIdentifier: 'sigma-redis',
+    params: [
+        //Operation parameters
+    ]
+}, function (error, response, redisClient) {
+    if (error) {
+        callback(error);
+    } else {
+        //redisClient.quit();
+    }
+});
+```
+
+### Perform operations on multiple keys using single Sigma operation
+
+Most of the Redis operations are restricted to interact with only one key inside a single operation. 
+With Sigma IDE, now you can perform same redis operation on multiple of keys using single Sigma operation. 
+
+For an example redis `get` operation returns value of corresponding key while Sigma allows
+you to fetch values of multiple keys using single Sigma get operation.
+
+<p align="center">
+  <img style="Padding:20px" width="400" src="./images/elasticache/ecache_multikey_operation.png">
+</p>
+
+```
+redis.get({
+    clusterIdentifier: 'sigma-redis',
+    params: [
+        'key1', 'key2', 
+    ]
+}, function (error, response, redisClient) {
+    if (error) {
+        callback(error);
+    } else {
+        //redisClient.quit();
+    }
+});
+```
+
+### Return Values 
+
+For every executed operation Sigma will return following attributes in its callback function.
+
+ - `error`: This object will contain error if any. Otherwise it will be `null`
+ - `response`: Response object of the operation. 
```

Comment:
. `[See below](add a link to the "Response Format" heading) for the response object structure.`